crates/bevy_math/src/bounding/bounded2d.rs

```diff
@@ -0,0 +1,284 @@
+use super::BoundingVolume;
+use crate::prelude::Vec2;
+
+/// A trait with methods that return 2d bounded volumes for a shape
+pub trait Bounded2d {
+    /// Get an axis-aligned bounding box for the shape with the given translation and rotation
+    fn aabb_2d(&self, translation: Vec2, rotation: f32) -> Aabb2d;
+    /// Get a bounding circle for the shape
+    fn bounding_circle(&self, translation: Vec2) -> BoundingCircle;
+}
+
+/// A 2D axis-aligned bounding box, or bounding rectangle
+pub struct Aabb2d {
+    /// The minimum point of the box
+    pub min: Vec2,
+    /// The maximum point of the box
+    pub max: Vec2,
+}
+
+impl BoundingVolume for Aabb2d {
+    type Position = Vec2;
+    type Padding = (Vec2, Vec2);
```

Comment:
Taking a tuple as an argument like `aabb.padded((min, max))` feels slightly odd and implicit to me.

Is it particularly useful to have the min and max padding be separate like this, or could it just be a single vector that uniformly expands/shrinks the AABB in all directions?

I suppose one case where it's useful is something like bevy_xpbd's AABB expansion based on velocity, but it's quite trivial to set min and max separately when necessary like `aabb.min -= something`.

Reply:
Shapecasting also requires offset padding like this (if the shape's center isn't 0, 0). But maybe we need to find a better solution for this, since circles/spheres could also be offset :thinking: